core/src/main/scala/org/scalarelational/datatype/DataType.scala

```diff
@@ -6,122 +6,133 @@ import org.powerscala.enum.{EnumEntry, Enumerated}
 import org.powerscala.reflect._
 import org.scalarelational.column.WrappedString
 import org.scalarelational.column.property.{IgnoreCase, NumericStorage}
-import org.scalarelational.model.{ColumnLike, Column}
+import org.scalarelational.model.ColumnLike
 
 /**
  * @author Matt Hicks <matt@outr.com>
  */
 trait DataType[T] {
-  def sqlType(column: ColumnLike[T]): String
+  def sqlType(column: ColumnLike[_]): String
+  def toSQLType(column: ColumnLike[_], value: T): Any
+  def fromSQLType(column: ColumnLike[_], value: Any): T
+}
+
+object DataTypeGenerators {
+  def option[T](implicit dt: DataType[T]): DataType[Option[T]] = new DataType[Option[T]] {
+    def sqlType(column: ColumnLike[_]) = dt.sqlType(column)
 
-  def toSQLType(column: ColumnLike[T], value: T): Any
+    def toSQLType(column: ColumnLike[_], value: Option[T]): Any = {
+      value match {
+        case None => null
```

Comment:
You can't just use "= null" in comparison...you have to use "IS NULL" or "IS NOT NULL".